resources.go

```diff
@@ -333,6 +337,16 @@ func Provider() tfbridge.ProviderInfo {
 			// DataBricks
 			"azurerm_databricks_workspace": {Tok: azureResource(azureDataBricks, "Workspace")},
 
+			// DataFactory
+			"azurerm_data_factory":                           {Tok: azureResource(azureDataFactory, "Factory")},
```

Comment:
I wonder if we'd be better mapping this one as `Factory`, but removing that prefix from all of the other resources?

Reply:
changed as requested!

Reply:
This looks good to merge, pending CI.